pkg/ebpf/c/tracee.bpf.c

```diff
@@ -6289,3 +6313,836 @@ int tc_ingress(struct __sk_buff *skb)
 {
     return tc_probe(skb, true);
 }
+
+// NOTE: new network code (works in <= 5.6 as well)
+
+// There are multiple ways to follow ingress/egress for a task. One way is to
+// try to track all flows within network interfaces and keep a map of addresses
+// tuples and translations. OR, sk_storage and socket cookies might help in
+// understanding which sock/sk_buff context the bpf program is dealing with but,
+// at the end, the need is always to tie a flow to a task (specially when
+// hooking ingress skb bpf programs, when the current task is a kernel thread
+// most of the times).
+
+// Unfortunately that gets even more complicated in older kernels: the cgroup
+// skb programs have almost no bpf helpers to use, and most of common code
+// causes verifier to fail. With that in mind, this approach uses a technique of
+// kprobing the function responsible for calling the cgroup/skb programs.
+
+// All the work, that should be done by the cgroup/skb programs in the common
+// case, in this case is done by this kprobe/kretprobe hook logic (right before
+// and right after the cgroup/skb program runs). By doing that, all the data
+// that cgroup/skb programs need to use is already placed in a map.
+
+// Obviously this has some cons: this kprobe->cgroup/skb->kretprobe execution
+// flow does not have preemption disabled, so the map used in between the 3
+// hooks need to use something that is available to all 3 of them.
+
+// At the end, the logic is simple: every time a socket is created an inode is
+// also created. The task owning the socket is indexed by the socket inode so
+// everytime this socket is used we know which task it belongs to (specially
+// during ingress hook).
+
+//
+// network helper functions
+//
+
+static __always_inline bool is_socket_supported(struct socket *sock)
+{
+    // clang-format off
+    struct sock *sk = (void *) BPF_READ(sock, sk);
```

Comment:
I'm planning to get rid of these helper functions that have a solo purpose. It takes more time for you to find if there is a helper available than to actually read the struct member/field you need. I plan to keep the more complex helpers (like the ones calculating member offsets because of bitfield relocations, and things that really deserve a helper).